test/realtime/channelgroup.test.js

```diff
@@ -531,5 +517,80 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
           closeAndFinish(done, [realtime1, realtime2, realtime3], err);
         });
     });
+
+    it('does not conflict with regular use of channel', function (done) {
+      const prefix = utils.cheapRandStr();
+      const activeChannelName = `${prefix}:active`;
+      const consumerGroupName = `${prefix}:testgroup`;
+      let realtime1 = helper.AblyRealtime({ clientId: 'testclient' });
+      let realtime2 = helper.AblyRealtime({ clientId: 'consumer' });
+
+      // connect
+      Promise.all([
+        new Promise((resolve) => realtime1.connection.on('connected', resolve)),
+        new Promise((resolve) => realtime2.connection.on('connected', resolve)),
+      ])
+        .then(async () => {
+          // create 2 consumers in one group
+          const consumer = realtime2.channelGroups.get(`${prefix}:.*`, {
+            activeChannel: activeChannelName,
+            consumerGroup: { name: consumerGroupName },
+          });
+
+          let events = 0;
+          let received = [null, null];
+          let message1 = new Promise((resolve) => (received[0] = resolve));
+          let message2 = new Promise((resolve) => (received[1] = resolve));
+          await consumer.subscribe((channel, msg) => {
+            received[events]();
+            events++;
+            expect(msg.data).to.equal('test data', 'Unexpected msg text received');
+            expect(channel).to.equal(`${prefix}:channel`, 'Unexpected channel name');
+          });
+          
+          // publish active channels
+          const channelName = `${prefix}:channel`;
+          let activeChannel = realtime1.channels.get(activeChannelName);
+          await activeChannel.attach();
+          activeChannel.publish('event0', { active: [channelName] });
+
+          // wait for the consumer to appear in the group
+          await waitForConsumers(realtime1.channels.get(consumerGroupName), 1);
```

Comment:
out of interest, why do we expect a consumer to appear in **realtime1**, if we've created a consumer previously on **realtime2** `const consumer = realtime2.channelGroups.get`?

Reply:
oh, I see, `waitForConsumers` actually checks for presence in channel, so it's cross-client. makes sense now.
maybe a slight name change to `waitForConsumers` would be good in a separate PR? to make it more clear for what it waits.